### PR TITLE
Fix transcript edit page

### PR DESCRIPTION
**Pull request by calvinlu3**
Fixes https://github.com/oncokb/oncokb-pipeline/issues/613

### CODE REVIEW

File: src/main/webapp/app/entities/alteration/alteration-update.tsx [hunk 1]
```diff
@@ -60,7 +61,7 @@ export const AlterationUpdate = (props: IAlterationUpdateProps) => {
       props.alterationEntity.genes && props.alterationEntity.genes.length > 0
         ? props.alterationEntity.genes.map(gene => ({
             label: gene.hugoSymbol,
-            value: gene.id,
+            value: gene,
```
Comment (bprize15) [inline src/main/webapp/app/entities/alteration/alteration-update.tsx:64]: I would stick with the id for value. If for some reason the reference changes we could end up in a weird state. If you do need to attach the actual gene, could you do it with another property?

File: src/main/webapp/app/entities/transcript/transcript-update.tsx [hunk 2]
```diff
@@ -46,6 +44,22 @@ export const TranscriptUpdate = (props: ITranscriptUpdateProps) => {
...
+    if (transcriptEntity?.gene) {
+      setGeneValue({
+        value: transcriptEntity.gene,
+        label: transcriptEntity.gene.hugoSymbol,
+        synonyms: transcriptEntity.gene.synonyms ?? [],
+      });
+    }
+    if (transcriptEntity?.ensemblGene) {
+      setEnsemblGeneValue({
+        value: transcriptEntity.ensemblGene,
```
Comment (bprize15) [inline src/main/webapp/app/entities/transcript/transcript-update.tsx:57]: Same comment as above about using an id as value

File: src/main/webapp/app/shared/select/EnsemblGeneSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    if (searchWord) {
+      result = await searchEnsemblGenes({ query: searchWord, page: page - 1, size: ITEMS_PER_PAGE, sort: [sortParameter] });
+    } else {
+      result = await getEnsemblGenes({ page: page - 1, size: ITEMS_PER_PAGE, sort: [sortParameter] });
+    }
+
+    options =
+      result?.data?.map(
+        (entity: IEnsemblGene): EnsemblGeneSelectOption => ({
+          value: entity,
```
Comment (bprize15) [inline src/main/webapp/app/shared/select/EnsemblGeneSelect.tsx:43]: Same comment as above, prefer id as value

File: src/main/webapp/app/shared/select/GeneSelect.tsx [hunk 4]
```diff
@@ -16,7 +16,7 @@ export interface IGeneSelectProps<IsMulti extends boolean> extends SelectProps<G
 const sortParameter = getEntityPaginationSortParameter(DEFAULT_ENTITY_SORT_FIELD[ENTITY_TYPE.GENE] ?? '', DEFAULT_SORT_DIRECTION);
 
 export interface GeneSelectOption {
-  value?: number;
+  value?: IGene;
```
Comment (bprize15) [inline src/main/webapp/app/shared/select/GeneSelect.tsx:19]: Same comment

File: src/main/webapp/app/entities/transcript/transcript-update.tsx [hunk 5]
```diff
@@ -54,9 +68,8 @@ export const TranscriptUpdate = (props: ITranscriptUpdateProps) => {
       ...transcriptEntity,
       ...values,
       flags: mapIdList(values.flags),
-      ensemblGene: ensemblGenes.find(it => it.id.toString() === values.ensemblGeneId.toString()),
-      // TYPE-ISSUE: Is selectedGeneId a number or a string?
-      gene: genes.find(it => it.id.toString() === selectedGeneId?.toString()),
+      ensemblGene: ensemblGeneValue ? { id: ensemblGeneValue.value } : undefined,
+      gene: geneValue ? { id: geneValue.value } : undefined,
```
Comment (calvinlu3) [inline src/main/webapp/app/entities/transcript/transcript-update.tsx:72]: @bprize15 Addressed your comments. I actually realized that we don't need to pass in the entire gene object eventhough TranscriptDTO is asking for a `Gene` type. You can just pass in an object with id set to the gene id. 
Actually makes everything simpler.

**Review by bprize15 — APPROVED** [review]
LGTM

**Pull request merged**